### PR TITLE
Create a custom fake simple plugin for testing

### PR DESCRIPTION
**Pull request by VaiTon**
Using a real plugin means:
1. we need to be online during testing
2. if the plugin has a different version from rizin
   on the machine often it won't event compile

This commit introduces a fake plugin in the
simpleplugin directory. During testing, an HTTP
server is spawned. The .tar.gz is created and
served on-the-fly to every test that asks for it.

### CODE REVIEW

File: simpleplugin/plugin.c [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  return true;
+}
+
+static bool rz_cmd_fini(RzCore *core) {
+  // no-op plugin, does nothing
+  RZ_LOG_INFO("Finalizing simple plugin\n");
+  return true;
+}
+
+RzCorePlugin rz_core_plugin_jsdec = {
```
Comment (wargio) [inline simpleplugin/plugin.c:24]:
```suggestion
RzCorePlugin rz_core_plugin_example = {
```

File: simpleplugin/plugin.c [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+#ifdef _MSC_VER
+#define _RZ_API __declspec(dllexport)
+#else
+#define _RZ_API __attribute__((visibility("default")))
+#endif
+
+#ifndef CORELIB
+_RZ_API RzLibStruct rizin_plugin = {
+    .type = RZ_LIB_TYPE_CORE,
+    .data = &rz_core_plugin_jsdec,
```
Comment (wargio) [inline simpleplugin/plugin.c:41]:
```suggestion
    .data = &rz_core_plugin_example,
```

File: pkg/utils_test.go [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+func TestGetMajorMinorVersion(t *testing.T) {
+	tests := []struct {
+		version string
+		want    string
+	}{
+		{"1.2.3", "1.2"},
+		{"10.20.30", "10.20"},
+		{"0.1.2", "0.1"},
+		{"5.6", "5.6"}, // Edge case with no patch version
```
Comment (wargio) [inline pkg/utils_test.go:13]: add another edge case. you are asking for `0.4` but you have only major version, example `1`

File: simpleplugin/plugin.c [hunk 4]
```diff
@@ -0,0 +1,44 @@
+// SPDX-FileCopyrightText: 2018-2023 Giovanni Dante Grazioli <deroad@libero.it>
+// SPDX-License-Identifier: BSD-3-Clause
```
Comment (wargio) [inline simpleplugin/plugin.c:2]: use as license LGPL and add your name.

File: simpleplugin/plugin.c [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+#include "rz_util/rz_log.h"
```
Comment (wargio) [inline simpleplugin/plugin.c:4]:
```suggestion
```

File: pkg/utils_test.go (hunk 3 above)
Comment (VaiTon) [reply pkg/utils_test.go:13]: I'm not quite sure I understand what you're asking. Something like `{"1", "1.0"}`?

Comment (wargio) [reply pkg/utils_test.go:13]: Yes.

**Comment by wargio** [comment]
I'll try to test this today and merge it
